src/timely-util/src/order.rs

```diff
@@ -150,6 +164,33 @@ impl Extrema for i32 {
     }
 }
 
+impl Extrema for Uuid {
+    fn minimum() -> Self {
+        Self::nil()
+    }
+    fn maximum() -> Self {
+        Self::from_bytes([0xff; 16])
+    }
+}
+
+pub trait Sequence {
```

Comment:
If we're going to define such a trait we should make look like the std one (which is currently unstable) https://doc.rust-lang.org/std/iter/trait.Step.html

Reply:
Makes sense, will update to follow that API